fix: bundle install logic

0xpatrickdev opened the pull request:
- see #4 

### Changes

- ensure proposal does not produce any installations, fixing behavior where - 1) the chain would result in 2 installations of the same bundle 2) the agoricNames.installation ref can only hold one value, and would be overriden for previous installations

- removes agoricNamesAdmin from proposal, but leaves in permit as it's needed in  https://github.com/Agoric/agoric-sdk/blob/5387e35d7826329e6d93b3e80b3ed1779887ba82/packages/inter-protocol/src/proposals/startPSM.js#L128



### Testing

```bash
2023-10-26T17:16:49.268Z block-manager: block 72426 begin
2023-10-26T17:16:49.271Z block-manager: block 72426 commit
2023-10-26T17:16:50.281Z block-manager: block 72427 begin
2023-10-26T17:16:50.302Z SwingSet: vat: v1: evaluateBundleCap { manifestBundleRef: { bundleID: 'b1-4c34c89b707bc8ece5a41e97e6a354081f7ae8a40391f1462848348613dd1218dcce574b3e30901a9825a966cb85bda6a92ba9f9ce9ba325e4c475f9a678b930' }, exportedGetManifest: 'getManifestForPsm', vatAdminSvc: Promise [Promise] {} }
2023-10-26T17:16:50.957Z SwingSet: vat: v1: execute { exportedGetManifest: 'getManifestForPsm', behaviors: [ 'INVITE_PSM_COMMITTEE_MANIFEST', 'PSM_MANIFEST', 'getManifestForPsm', 'getManifestForPsmGovernance', 'inviteCommitteeMembers', 'inviteToEconCharter', 'makeAnchorAsset', 'startEconCharter', 'startPSM' ] }
2023-10-26T17:16:50.960Z SwingSet: vat: v1: coreProposal: makeAnchorAsset
2023-10-26T17:16:50.960Z SwingSet: vat: v1: coreProposal: startPSM
2023-10-26T17:16:52.761Z block-manager: block 72427 commit
2023-10-26T17:16:52.915Z block-manager: block 72428 begin
2023-10-26T17:16:53.167Z SwingSet: vat: v28: provisionPool notified of new asset Object [Alleged: USDC brand] {}
2023-10-26T17:16:53.178Z SwingSet: vat: v43: ----- WltFct.4  13 registering asset USDC
2023-10-26T17:16:54.664Z SwingSet: vat: v28: provisionPool balance update { brand: Object [Alleged: USDC brand] {}, value: 0n }
2023-10-26T17:16:56.372Z SwingSet: vat: v58: PSM Starting Object [Alleged: USDC brand] {} { denominator: { brand: Object [Alleged: IST brand] {}, value: 1_000_000n }, numerator: { brand: Object [Alleged: USDC brand] {}, value: 1_000_000n } }
2023-10-26T17:16:56.950Z SwingSet: vat: v15: charter: adding instance psm-IST-USDC
2023-10-26T17:16:57.017Z block-manager: block 72428 commit
2023-10-26T17:16:57.198Z block-manager: block 72429 begin
2023-10-26T17:16:57.237Z block-manager: block 72429 commit
2023-10-26T17:16:58.210Z block-manager: block 72430 begin

```

0xpatrickdev committed:
fix: bundle install logic

- ensure proposal does not produce any installations, fixing behavior where -
  1) the chain would result in 2 installations of the same bundle
  2) the agoricNames.installation ref can only hold one value, and would be overriden for previous installations

- removes agoricNamesAdmin from proposal, but leaves in permit as it's needed in  https://github.com/Agoric/agoric-sdk/blob/5387e35d7826329e6d93b3e80b3ed1779887ba82/packages/inter-protocol/src/proposals/startPSM.js#L128

dckc reviewed (changes requested):
as we discussed, please go ahead and take out the `vatAdminSvc` permit by setting `restoreRef = (() => {})` and such.

thanks for the clues, @michaelfig

0xpatrickdev committed:
fix: ensure bundles are not installed in defaultRestoreRef flow

- skips `defaultRestoreRef` instantiation, which calls `E(zoe).installBundleID...`
- sets `restoreRef` to a noop function, as `startPSM.js` relies on its existence
- preserves `installKeys` key in manifest, as `startPSM.js` relies on its existence

0xpatrickdev commented:
Thank you both!  Changes are in 49b5c7a. If we are to remove `vatAdminSvc` completely from the permit, I might need more  pointers around bundleCap and manifestNS. 

The logs, after a local dry-run, seem to indicate success:

```bash
2023-10-26T21:01:03.982Z block-manager: block 780 commit
2023-10-26T21:01:04.922Z block-manager: block 781 begin
2023-10-26T21:01:04.933Z SwingSet: vat: v1: evaluateBundleCap { manifestBundleRef: { bundleID: 'b1-4c34c89b707bc8ece5a41e97e6a354081f7ae8a40391f1462848348613dd1218dcce574b3e30901a9825a966cb85bda6a92ba9f9ce9ba325e4c475f9a678b930' }, exportedGetManifest: 'getManifestForPsm', vatAdminSvc: Promise [Promise] {} }
2023-10-26T21:01:05.437Z SwingSet: vat: v1: execute { exportedGetManifest: 'getManifestForPsm', behaviors: [ 'INVITE_PSM_COMMITTEE_MANIFEST', 'PSM_MANIFEST', 'getManifestForPsm', 'getManifestForPsmGovernance', 'inviteCommitteeMembers', 'inviteToEconCharter', 'makeAnchorAsset', 'startEconCharter', 'startPSM' ] }
2023-10-26T21:01:05.438Z SwingSet: vat: v1: coreProposal: makeAnchorAsset
2023-10-26T21:01:05.438Z SwingSet: vat: v1: coreProposal: startPSM
2023-10-26T21:01:07.049Z block-manager: block 781 commit
2023-10-26T21:01:07.121Z block-manager: block 782 begin
2023-10-26T21:01:07.320Z SwingSet: vat: v28: provisionPool notified of new asset Object [Alleged: USDC brand] {}
2023-10-26T21:01:07.325Z SwingSet: vat: v43: ----- WltFct.4  13 registering asset USDC
2023-10-26T21:01:08.705Z SwingSet: vat: v28: provisionPool balance update { brand: Object [Alleged: USDC brand] {}, value: 0n }
2023-10-26T21:01:10.372Z SwingSet: vat: v58: PSM Starting Object [Alleged: USDC brand] {} { denominator: { brand: Object [Alleged: IST brand] {}, value: 1_000_000n }, numerator: { brand: Object [Alleged: USDC brand] {}, value: 1_000_000n } }
2023-10-26T21:01:11.192Z SwingSet: vat: v15: charter: adding instance psm-IST-USDC
2023-10-26T21:01:11.261Z block-manager: block 782 commit
2023-10-26T21:01:11.360Z block-manager: block 783 begin
2023-10-26T21:01:11.364Z block-manager: block 783 commit
2023-10-26T21:01:12.372Z block-manager: block 784 begin
```

gibson042 commented on gov-start-usdc-psm.js:
```diff
@@ -162,28 +139,14 @@ const overrideManifest = {
+    const options = await shallowlyFulfilled(rawOptions);
 
+    // @skip, these bundles are already installed
     // Publish the installations for behavior dependencies.
```
```suggestion
    // All dependency bundles are already installed, so we ignore `installations` from the getManifest response.
```

gibson042 commented on gov-start-usdt-psm.js:
```diff
@@ -162,28 +139,14 @@ const overrideManifest = {
+    const options = await shallowlyFulfilled(rawOptions);
 
+    // @skip, these bundles are already installed
     // Publish the installations for behavior dependencies.
```
```suggestion
    // All dependency bundles are already installed, so we ignore `installations` from the getManifest response.
```

gibson042 reviewed (approved):
It's unfortunate to see `vatAdminSvc` remain, but neither @dckc nor I could come up with a good way to remove it that would be in scope here.

As for everything else, it makes sense to me—reiterating here outside-in in case I've misunderstood something:
1. `behavior` for gov-start-{usdc,usdt}-psm.js proposal scripts does not need to install any bundles because everything they need is already available.
2. Therefore the proposal scripts can ignore `installations` from the getManifest call ("getManifestForPsm").
3. They can also provide a no-op `restoreRef` to that call, which only uses it to resolve "psm" and "mintHolder" `installKeys` that for the same reason are no longer present in its input.
4. Accordingly, the proposal scripts no longer need to deconstruct `installation` or `agoricNamesAdmin` from their provided powers (the latter having been used for looking up the installation admin, but still remaining in their permits for `makeAnchorAsset` and `startPSM`).

0xpatrickdev committed:
Update gov-start-usdt-psm.js

Co-authored-by: Richard Gibson <richard.gibson@gmail.com>

0xpatrickdev committed:
Update gov-start-usdc-psm.js

Co-authored-by: Richard Gibson <richard.gibson@gmail.com>
